File: 30diff/119probleminc/DigitPowerSum.java
import java.util.ArrayList; 

public class DigitPowerSum{
    public static void main(String[] commandLine) {
        long start = System.currentTimeMillis();
        boolean done = false; 
        ArrayList<Integer> a = new ArrayList<Integer>();
        for(int n = 10; !done; n++, done = a.size() == 30) { if (inSequence(n)) a.add(n); }
        System.out.println(a);
        long end = System.currentTimeMillis();
        System.out.println(end-start);
    }
    
    private static boolean inSequence(int n) {
        int sum = sumDigits(n);
        if (sum == 1) return false; 
        for(int power = 1; Math.pow(sum, power) <= n; power++) { if (Math.pow(sum,power) == n) return true; }
        return false; 
    }
    
    private static int sumDigits(int n) {
        if ( n < 10 ) return n;
        else return sumDigits(n / 10) + sumDigits(n % 10);
    }
}

File: 30diff/119probleminc/digitPowerSum.py
import time
import math 
def solution():
    done = False
    test = 10
    a = []
    while not done:
        if inSequence(test):
            a.append(test)
        test += 1
        done = len(a) == 30
    print a
    
def inSequence(n):
    sum = sumDigits(n)
    if sum == 1:
        return False 
    exponent = str(math.log(n, sum))
    return exponent[len(exponent)-2:] == ".0"
    # power = 1
    # while sum ** power <= n:
        # if sum**power == n:
            # return True 
        # power+=1
    # return False 
    
def sumDigits(n):
    if n < 10:
        return n
    else:
        return sumDigits(n/10) + sumDigits(n % 10)

start = time.time()
solution()
end = time.time()
print end-start 